addons/xterm-addon-serialize/src/SerializeAddon.ts

```diff
@@ -469,3 +512,154 @@ interface ISerializeOptions {
   excludeModes?: boolean;
   excludeAltBuffer?: boolean;
 }
+
+interface IHtmlSerializeOptions {
+  scrollback: number;
+  onlySelection: boolean;
+}
+
+export class HTMLSerializeHandler extends BaseSerializeHandler {
+  private _currentRow: string = '';
+
+  private _htmlContent = '';
+
+  private _inverseStyle = 'color: #000000; background-color: #BFBFBF;';
+  private _blinkStyle = 'text-decoration: blink;';
+  private _boldStyle = 'font-weight: bold;';
+  private _italicStyle = 'font-style: italic;';
+  private _underlineStyle = 'text-decoration: underline;';
+  private _strikethroughStyle = 'text-decoration: line-through;';
+  private _invisibleStyle = 'visibility: hidden;';
+  private _dimStyle = 'opacity: 0.5;';
+
+  private _colors: IColorSet;
+
+  constructor(
+    buffer: IBuffer,
+    private readonly _terminal: Terminal
+  ) {
+    super(buffer);
+
+    // https://github.com/xtermjs/xterm.js/issues/3601
+    this._colors = (_terminal as any)._core._colorManager.colors;
+  }
+
+  private _padStart(target: string, targetLength: number, padString: string): string {
+    targetLength = targetLength >> 0;
+    padString = String(typeof padString !== 'undefined' ? padString : ' ');
```

Comment:
You can use `padString !== undefined` here, I stopped caring about people who want to override `undefined` years ago 😉. Also I think the `String()` is redundant and this could be simplified:

```suggestion
    padString = padString ?? ' ';
```